dfir_rs/tests/surface_loop.rs

```diff
@@ -243,3 +243,99 @@ pub fn test_flo_repeat_n_multiple_nested() {
         &*collect_ready::<Vec<_>, _>(&mut result2_recv)
     );
 }
+
+#[multiplatform_test(test, wasm, env_tracing)]
+pub fn test_flo_repeat_kmeans() {
+    const POINTS: &[[i32; 2]] = &[
+        [-210, -104],
+        [-226, -143],
+        [-258, -119],
+        [-331, -129],
+        [-250, -69],
+        [-202, -113],
+        [-222, -133],
+        [-232, -155],
+        [-220, -107],
+        [-159, -109],
+        [-49, 57],
+        [-156, 52],
+        [-22, 125],
+        [-140, 168],
+        [-118, 89],
+        [-93, 133],
+        [-101, 80],
+        [-145, 79],
+        [187, 36],
+        [208, -66],
+        [142, 5],
+        [232, 41],
+        [91, -37],
+        [132, 16],
+        [248, -39],
+        [158, 65],
+        [108, -41],
+        [171, -121],
+        [147, 5],
+        [192, 58],
+    ];
+    const CENTROIDS: &[[i32; 2]] = &[[-50, 0], [0, 0], [50, 0]];
+
+    let (result_send, mut result_recv) = dfir_rs::util::unbounded_channel::<_>();
+
+    let mut df = dfir_syntax! {
+        init_points = source_iter(POINTS) -> map(std::clone::Clone::clone);
+        init_centroids = source_iter(CENTROIDS) -> map(std::clone::Clone::clone);
+        loop {
+            batch_points = init_points -> batch();
```

Comment:
It might be nice to have a slightly different syntax to distinguish windowing / un-windowing operators, but this is fine for now.

Reply:
I agree